### DISCUSSION TITLE
Can repo release be made, which would include sub-Worktrees, without duplicating their content?

### DISCUSSION THREAD

**Discussion by irvnriir** [git-for-windows/git#4326] [Q&A]
When im creating Commit with the parent folder, it seems that the subtrees are ignored so i need a separate script for Checkouting them when Cloning . Also Github uses Tags for releasing, which are attached to Commits .

For Github, it seems like i need to temporally remove the Worktrees, then Commit all their content and release the result . But would this duplicate the content in .git ? Considering that the most of repo consists of sub-Worktrees, thats a huge storing issue .

Git Submodules seem to each require a separate .git, so they don't appear relevant here .

**Comment by dscho** [comment]
> When im creating Commit with the parent folder, it seems that the subtrees are ignored so i need a separate script for Checkouting them when Cloning .

I am sure that I don't understand correctly. Would you provide a small script to illustrate the issue?

**Reply by irvnriir** [reply]
> When im creating Commit with the parent folder, it seems that the subtrees are ignored

Im not sure what is hard to understand here, maybe you tell how you see it ?

**Reply by dscho** [reply]
What is the "parent folder"? The parent folder of the folder containing `.git`? And in the subtrees (which I _think_ are what I would call "subdirectories" or "subfolders", as Git's glossary refers to trees as [something that seems totally not applicable here](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddeftreeatree)), do they contain `.git` items? Have you `git add`ed the files in them? Git cannot commit directories themselves, Git only tracks files, symbolic links and submodules, not directories.

So yes, I find the original question super hard and non-obvious to understand.

**Reply by irvnriir** [reply]
A subtree here is a short version of "Worktree inside the main Worktree" , parent folder is the main (parent) Worktree .

**Reply by irvnriir** [reply]
Doing Commit with a folder means committing all it's content .

**Comment by irvnriir** [comment]
my bad, i now tried running `git stage --no-ignore-removal` directly, and discovered that the subtrees are actually get added like this:
```
commit <<_>>
Author: <<_>>
Date:   <<_>>

    <<last Commit Message>>

<<changes in the last Commit>>
```

ty @dscho for the idea .

though the question stays . Git showed a hint saying that content of the subtree won't be Cloned, so, assuming that Git intentionally doesn't store content of the subtree and in terms of avoiding external scripts, can a different Branch in repository be a target for a Submodule in the same repository, somehow ?

**Comment by irvnriir** [comment]
i'll also be grateful if someone can suggest adding the subtees support or relevant documentation to git, since it's repo doesn't have Issues and i assume that the git-for-windows are not adding features above linux's git .
